File: test/lira_output/ts/features/f7_interface/greeter.ts
// module f7_interface
export interface Greeter {
  greet(name: string): string;
}

export class Person implements Greeter {
  public name: unknown;
  constructor(name: string) {
    this.name = name;
  }
  public greet(name: string): string {
    return this.name;
  }
}

File: test/lira_output/py/features/f7_interface/greeter.py
# module f7_interface
from __future__ import annotations
from typing import Protocol

class Greeter(Protocol):
    def greet(self, name: str) -> str:
        ...

class Person(Greeter):
    name: str

    def __init__(self, name: str) -> None:
        self.name = name


    def greet(self, name: str) -> str:
        return self.name
